app/questionnaire/questionnaire_store_updater.py

```diff
@@ -505,7 +512,9 @@ def remove_dependent_blocks_and_capture_dependent_sections(self) -> None:
             for block_id in blocks_to_remove:
                 location = Location(
                     section_id=section_id,
-                    list_item_id=list_item_id,
+                    list_item_id=None
```

Comment:
I reckon the change I've done for looping-4 would work here after all, because then the dynamic-answers block won't have a list-item-id associated to it, which was the problem I had too, so then you shouldn't need to manually modify the location here

Reply:
I think the only change that would be needed is the bit in questionnaire_schema which does this 
```
     for block_answer_id in answer_ids_for_block:
            self._answer_dependencies_map[block_answer_id] |= {
                self._get_answer_dependent_for_block_id(
                    block_id=block_id, for_list=list_name
                )
                if self.is_block_in_repeating_section(block_id)
                # non-repeating blocks such as dynamic-answers could depend on the list
                else self._get_answer_dependent_for_block_id(block_id=block_id)
            }
```

Reply:
What an improvement, great stuff. Thanks.